### ISSUE TITLE
Widget Rule definitions should allow day, day of month, month selection for all frequencies

### ISSUE THREAD

**Issue by humbled** [django-recurrence/django-recurrence#45]
### tl;dr
The dateutil docs give examples like:
```
list(rrule(YEARLY, interval=4, count=3, bymonth=11,
               byweekday=TU, bymonthday=(2,3,4,5,6,7,8),
               dtstart=parse("19961105T090000")))
[datetime.datetime(1996, 11, 5, 9, 0),
 datetime.datetime(2000, 11, 7, 9, 0),
 datetime.datetime(2004, 11, 2, 9, 0)]
```
but we can't specify months AND days of week in the same rule in the django-recurrence widget

### Detail
I need to specify a range for certain days of the week, but only for particular months of the year (for seasonal activities). I can achieve an example of this in python with:

    list(rrule(YEARLY, bymonth=(1,2,3,7,9), byweekday=(SA,SU), count=10))

(list and count included for readability of output)

As awesome as django-recurrence is, each frequency tab (eg Annually, Weekly) only supports options relative to that type, so Annually doesn't let me specify for which weekdays, for example. I can add another rule, but that is a logical OR which means that dates are included if they match the first rule or the second rule.

One way around this would be for me to use an Exclude, but this seems to only be a single day in the month (when I choose Annually and my months). Let me know if I'm missing something here.

Another way, which would be an enhancement to the widget. It seems from the dateutil docs that the Frequency selected is only relevant for *interval* and *bysetpos*. This means that any Frequency can have day of week, week of year, etc. So the widget would need reworking so that regardless of Annually / Monthly / Weekly / Daily selected, the month, day of month, and day of week selectors would always be visible.

Discussion welcome

**Comment by dominicrodger** [comment]
Hey @humbled - thanks for the bug report.

I think this boils down to the same basic problem as reported in #37 - that there are various rules expressible in dateutil which aren't expressible using the django-recurrence widget. I'm honestly not really sure how to handle this - without a fairly significant JS change (and possibly rewrite, per #4). There are a couple of other issues that require JS changes that we've also punted on (#14 and #39).

I'm open to patches that address any of these - bonus points if anyone can set up some basic JS tests for django-recurrence too (the majority of the code for django-recurrence is JS, but all our unit tests are for the Python).

Dom
